http-server-netty/src/main/java/io/micronaut/http/server/netty/MicronautHttpData.java

```diff
@@ -0,0 +1,781 @@
+                    chunk.lock.unlock();
+                }
+            }
+            chunks.add(new Chunk(size));
+        }
+        // add to the chunk
+        try {
+            if (channel == null) {
+                if (chunk.buf == null) {
+                    chunk.buf = buffer;
+                } else if (chunk.buf instanceof CompositeByteBuf composite) {
+                    composite.addComponent(true, buffer);
+                } else {
+                    chunk.buf = Unpooled.compositeBuffer()
+                        .addComponent(true, chunk.buf)
+                        .addComponent(true, buffer);
+                }
+            } else {
+                buffer.readBytes(channel, size, buffer.readableBytes());
+                buffer.release();
+                chunk.loadFromDisk(newChunkSize);
+            }
+            size = newSize;
+            if (newSize > definedSize && definedSize != 0) {
+                definedSize = newSize;
+            }
+        } finally {
+            chunk.lock.unlock();
+        }
+        if (last) {
+            completed = true;
+            if (channel != null) {
+                channel.close();
+            }
+        }
+    }
+
+    private ByteBuf mmapSegment(int index) throws IOException {
+        while (mmapSegments.size() <= index) {
+            mmapSegments.add(null);
+        }
+        ByteBuf segment = mmapSegments.get(index);
+        if (segment == null) {
+            segment = Unpooled.wrappedBuffer(
+                channel.map(FileChannel.MapMode.READ_ONLY, (long) index * MMAP_SEGMENT_SIZE, MMAP_SEGMENT_SIZE));
+            mmapSegments.set(index, segment);
+        }
+        return segment;
+    }
+
+    private void transferToDisk() throws IOException {
+        assert channel == null;
+
+        path = newTempFile();
+        channel = FileChannel.open(path, StandardOpenOption.READ, StandardOpenOption.WRITE);
+
+        for (Chunk chunk : chunks) {
+            if (chunk.buf != null) {
+                chunk.buf.getBytes(chunk.buf.readerIndex(), channel, chunk.offset, chunk.buf.readableBytes());
+            }
+        }
+        mmapSegments = new ArrayList<>();
+        for (Chunk chunk : chunks) {
+            if (chunk.lock.tryLock()) {
+                try {
+                    if (chunk.buf != null) {
+                        chunk.loadFromDisk(chunk.buf.readableBytes());
+                    }
+                } finally {
+                    chunk.lock.unlock();
+                }
+            } // if tryLock failed, the user already requested the chunk, we can't move it anymore.
+        }
+    }
+
+    private Path newTempFile() throws IOException {
+        Optional<File> location = factory.multipartConfiguration.getLocation();
+        if (location.isPresent()) {
+            return Files.createTempFile(location.get().toPath(), "FUp_", ".tmp");
+        } else {
+            return Files.createTempFile("FUp_", ".tmp");
```

Comment:
Just checking we are not going to be hit with another CVE for creating temp files, this uses the secure version so I assume we are ok

Reply:
sonar complained about this, but it's the same thing netty does, and openjdk 17 seems to create temp files with rwx------ permissions, so it should be fine